---
title: "Figure 4, Table S4, Table S5"
author: "James C. Kosmopoulos"
date: "`r Sys.Date()`"
output: github_document
editor_options:
  chunk_output_type: inline
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(warning = FALSE, message = FALSE, echo = TRUE)
```

# Load packages

```{r load-packages}
library(tidyverse); packageVersion("tidyverse")
library(ggforce); packageVersion("ggforce")
library(gggenes); packageVersion("gggenes")
```


# Load data and features
```{r load-data}
data_reform <- readRDS("../Data/stats_and_metadata.RDS")
depth_per_base <- readRDS("../Data/depth_per_base.RDS")
features <- readRDS("../Data/features_combined.RDS")
rect <- readRDS("../Data/features_absent.RDS")
```

# Table S4
```{r TableS4}
tableS4 <- features[ c("Genome", "contig",	"gene", "start", "stop", "start.relative",	"stop.relative", "frame", "phrog", "Method", "annot", "category")]
tableS4$sample <- sub("__.*", "", tableS4$Genome) # add a sample column
tableS4 <- tableS4[c(1, ncol(tableS4), 2:(ncol(tableS4)-1))] # move sample column to second position
tableS4 <- merge(tableS4, data_reform[c("Sample", "Environment")], by.x = "sample", by.y = "Sample")
col_order <- c("Genome", "sample", "Environment", setdiff(names(tableS4), c("Genome", "sample", "Environment")))
tableS4 <- tableS4[col_order]
write_csv(tableS4, file="../Tables/TableS4.csv")
head(tableS4)
```

# Table S5
```{r TableS5}
tableS5 <- depth_per_base[ c("Contig",	"Position",	"Depth",	"Relative.position",	"Read.sample",	"Read.sample.method",	"Genome.method",	"Genome")]
reads_paired_Ga0485184 <- data_reform[which(data_reform$Sample == "Ga0485184"),]$Reads_paired_reads[1]
reads_paired_Ga0485172 <- data_reform[which(data_reform$Sample == "Ga0485172"),]$Reads_paired_reads[1]
tableS5$Filtered.read.pairs.in.sample <- ifelse(tableS5$Read.sample == "Ga0485184", 
                                                reads_paired_Ga0485184, 
                                                ifelse(tableS5$Read.sample == "Ga0485172", 
                                                       reads_paired_Ga0485172, 
                                                       NA))
                             
tableS5$Depth.per.100M.reads <- depth_per_base$Depth.per.100M.reads
tableS5$Depth.normalized <- depth_per_base$Depth.normalized
write_csv(tableS5, file="../Tables/TableS5.csv")
head(tableS5)
```

# Figure 4
```{r Fig4, fig.height=4, fig.width=6.5}
# Define the offset for assembly B
offset_B <- 2789

# Define the range that is missing in assembly B
missing_range_start <- 6822 # Reversed
missing_range_end <- 6692 # Reversed
additional_missing_start <- 3905
additional_missing_end <- 3908

ggenes <- ggplot() +
  geom_gene_arrow(data = features, aes(xmin = start.relative, xmax = stop.relative, y = -0.5, fill = Genome.method)) +
  geom_line(data=depth_per_base %>%
              mutate(Depth.normalized = if_else(Depth.normalized < 0, 0, Depth.normalized)),
            aes(x=Relative.position, y=Depth.normalized, color=Read.sample.method)) +
  facet_grid(rows=vars(Genome.method), switch="both") +
  scale_y_continuous(sec.axis = sec_axis(~ . , name = "Genome assembly method", breaks = NULL, labels = NULL), position = "right",) +
  scale_color_manual(name= "Read sample method", values = c("Virome" = "#D81B60", "Mixed MG" = "#1E88E5")) +
  theme_linedraw() +
  theme(legend.position = "bottom") +
  ylab(expression(log[10]~normalized~read~depth)) +
  xlab("Genome position") +
  scale_fill_manual(name= "Method", values = c("Virome" = "#D81B60", "Mixed MG" = "#1E88E5")) +
  guides(fill="none") +
  annotate(
    "rect",
    xmin = missing_range_start,
    xmax = missing_range_end,
    ymin = -.75,
    ymax = max(depth_per_base$Depth.normalized),
    fill = "white",
    alpha = 1
  )
ggenes
ggsave("../Plots/Fig4.png", plot=ggenes, device = "png", width = 6.5, height = 4, units = "in", dpi = 600, scale = 1)
```

**Note that this figure was later modified, manually, to add highlighted regions and coloring for a selection of genes of interest.**



